### ISSUE TITLE
Optional Auth Decorator

### ISSUE THREAD

**Issue by HealYouDown** [dusktreader/flask-praetorian#172]
It would be great to see a decorator just like `auth_required` for `auth_optional`. If you then call `current_user` in the wrapped function, you get either None, if no token was supplied, or the user object.

Currently I have to use a hacky approch that adds an named argument to the wrapped function to solve this problem.

```
def auth_optional(method):
    """
    Decorator to check if a token exists and then adds an argument
    to the wrapped function called 'user'.
    User is either the database user object or None, if no token was supplied.
    """
    @wraps(method)
    def wrapper(*args, **kwargs):
        try:
            token = guard.read_token_from_header()
            _verify_and_add_jwt()
            user = current_user()
        except:
            user = None
        return method(*args, user=user, **kwargs)
    return wrapper
```

**Comment by dusktreader** [comment]
Sounds like a useful feature!

**Comment by dusktreader** [comment]
Resolved in merge request #216
